src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Completion/RazorDirectiveAttributeCommitManager.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion.Data;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.Editor.Razor.Completion
+{
+    internal class RazorDirectiveAttributeCommitManager : IAsyncCompletionCommitManager
+    {
+        public IEnumerable<char> PotentialCommitCharacters => new[] { '=' };
+
+        public bool ShouldCommitCompletion(IAsyncCompletionSession session, SnapshotPoint location, char typedChar, CancellationToken token)
+        {
+            if (session is null)
+            {
+                throw new ArgumentNullException(nameof(session));
+            }
+
+            if (!session.Properties.TryGetCompletionItemKinds(out _))
+            {
+                // There were no completions provided from our directive attribute completion provider(s).
+                return false;
+            }
+
+            // Directive attribute completion. This class is only ever called for our specific commit characters, allow the commit.
+
+            return true;
+        }
+
+        public CommitResult TryCommit(IAsyncCompletionSession session, ITextBuffer buffer, CompletionItem item, char typedChar, CancellationToken token)
+        {
+            // Do default behavior for the commit. This enables things like typing `=` and letting the Html completion engine finish off the attribute.
```

Comment:
Note: The default HTML experience auto-completes attribute to be suffixed with `=""` on tab/enter. To make that happen with our directive provider we'd need to add a lot of TextBuffer edit transaction magic here. It's something I want to add but am putting that experience off until other deliverables are complete.

As an alternative if users commit their directive attribute completions with a `=` it'll automagically complete the double quotes because of how the HTML editor works today.